clang/lib/3C/ProgramInfo.cpp

```diff
@@ -599,16 +599,14 @@ ProgramInfo::insertNewFVConstraint(FunctionDecl *FD, FVConstraint *NewC,
   if (ReasonFailed == "") return (*Map)[FuncName];
 
   // Error reporting
-  { // block to force DiagBuilder destructor and emit message
-    clang::DiagnosticsEngine &DE = C->getDiagnostics();
-    unsigned FailID = DE.getCustomDiagID(DiagnosticsEngine::Fatal,
-                                         "merging failed for %q0 due to %1");
-    const auto Pointer = reinterpret_cast<intptr_t>(FD);
-    const auto Kind = clang::DiagnosticsEngine::ArgumentKind::ak_nameddecl;
-    auto DiagBuilder = DE.Report(FD->getLocation(), FailID);
-    DiagBuilder.AddTaggedVal(Pointer, Kind);
-    DiagBuilder.AddString(ReasonFailed);
-  }
+  clang::DiagnosticsEngine &DE = C->getDiagnostics();
+  unsigned FailID = DE.getCustomDiagID(DiagnosticsEngine::Fatal,
+                                       "merging failed for %q0 due to %1");
+  const auto Pointer = reinterpret_cast<intptr_t>(FD);
+  const auto Kind = clang::DiagnosticsEngine::ArgumentKind::ak_nameddecl;
+  auto DiagBuilder = DE.Report(FD->getLocation(), FailID);
+  DiagBuilder.AddTaggedVal(Pointer, Kind);
+  DiagBuilder.AddString(ReasonFailed);
```

Comment:
Why not `<<` here?

Reply:
Because the pointer type needs a tag (<< is templated, tags are specific) and the default may not be what we want. I didn't check deeply to see what those two were, since there is no issue with using the local var here.

Reply:
I'd strongly prefer not to leave one occurrence unmigrated just because the local variable isn't causing a problem right now.  Generally speaking (maybe not likely in this particular case), someone could easily add another diagnostic later in the same scope and be unaware of the need to add a block, and the code would crash at runtime.  And our regression test coverage of many of the diagnostics is currently poor, even ignoring the fact that the tests that used the diagnostic verifier are currently XFAILed.

Fortunately, it looks like we can define the missing `<<` overload ourselves.  I'm doing so in my PR.

Reply:
Great, I'll wait until that in merged so I can use it in this PR.